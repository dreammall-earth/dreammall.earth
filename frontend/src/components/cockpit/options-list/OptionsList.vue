<template>
  <TransitionGroup name="options">
    <ul v-if="props.showIf" class="options-list">
      <slot />
    </ul>
  </TransitionGroup>
</template>

<script lang="ts" setup>
import { defineProps } from 'vue'

const props = defineProps<{
  showIf: boolean
}>()
</script>

<style scoped>
.options-list {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: calc(100% - 30px);
  background-color: var(--v-sidebar-background);
  display: flex;
  padding: 0;
  list-style: none;
}

.options-move, /* apply transition to moving elements */
.options-enter-active,
.options-leave-active {
  transition: all 0.3s ease;
}

.options-enter-from,
.options-leave-to {
  opacity: 0;
  transform: translateX(30px);
}

/* ensure leaving items are taken out of layout flow so that moving
   animations can be calculated correctly. */
.options-leave-active {
  position: absolute;
}
</style>
