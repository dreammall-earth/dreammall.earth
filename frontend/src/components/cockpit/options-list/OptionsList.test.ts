import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import OptionsList from './OptionsList.vue'

describe('OptionsList', () => {
  const Wrapper = (
    props = {
      isVisible: true,
    },
  ) => mount(OptionsList, { props })

  it('renders when visible', () => {
    const wrapper = Wrapper()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders when not visible', () => {
    const wrapper = Wrapper({ isVisible: false })
    expect(wrapper.element).toMatchSnapshot()
  })
})
