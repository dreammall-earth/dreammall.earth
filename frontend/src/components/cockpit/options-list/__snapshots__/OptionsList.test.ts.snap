// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OptionsList > renders when not visible 1`] = `
<transition-group-stub
  appear="false"
  css="true"
  data-v-01c78171=""
  name="options"
  persisted="false"
>
  <!--v-if-->
</transition-group-stub>
`;

exports[`OptionsList > renders when visible 1`] = `
<transition-group-stub
  appear="false"
  css="true"
  data-v-01c78171=""
  name="options"
  persisted="false"
>
  <ul
    class="options-list"
    data-v-01c78171=""
  >
    
    
  </ul>
</transition-group-stub>
`;
